WebRoot/js/zimbraAdmin/common/ZaTree.js

```diff
@@ -586,7 +586,8 @@ ZaTree.prototype._updateHistory =
 function (treeItem, isAddHistory, isShowInHistory) {
     var text = treeItem.getText();
     var dataItem = treeItem.getData("dataItem");
-    var path = this.getABPath(dataItem);
+    // Data taken from treeItem object is not going to be encoded, so always encode it before adding it to path
+    var path = AjxStringUtil.htmlEncode(this.getABPath(dataItem));
```

Comment:
nit: this feels wrong to be encoding the path here... we're putting this in a History object and not generating output here, no?  There seems to be numerous cases where the raw path is correct (for lookups, etc.) and the encoding should only be done on output, no?  Again, maybe I missed something.